### PR TITLE
Add onboarding flow

### PR DESCRIPTION
**Pull request by smalluban**
The `chrome.runtime.onInstalled` API does not work properly on Safari (as we expected), so the PR introduces the `version` property in options, which is checked to verify if the background process is run for the first time. We can use it in the future for update detection too.

It's still a work in progress, needs UI/UX for the page, and probably a more sophisticated condition to show the page to the user.

### CODE REVIEW

**Comment by smalluban** [comment]
@chrmod @philipp-classen Updated branch with the latest code. It is still not finished. Missing parts:
* Privacy Policy iframe feature
* CTA on popup panel when onboarding is skipped (closed tab, or skip in the process)
* Outro screen (I am not sure if this is ready to be implemented, as on desktop designs we still have lorem ipsums).

Important notice - the popup panel cannot be closed programmatically, so displaying the onboarding screen instead of the panel itself is no option. We need to create a CTA on the panel when Ghostery is not enabled. For now, I put there a simple placeholder link with `TODO` text ;)

File: src/background/onboarding.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+import { store } from 'hybrids';
+import Options from '/store/options.js';
+
+(async () => {
```
Comment (chrmod) [inline src/background/onboarding.js:14]: for a sake of the stacktrace it would be nice to name this function

File: src/background/onboarding.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  const options = await store.resolve(store.get(Options));
+  const version = chrome.runtime.getManifest().version;
+
+  if (!options.version) {
```
Comment (chrmod) [inline src/background/onboarding.js:18]: is this condition sufficient? shouldn't we retrigger the onboarding if the uses did not interact with it ?

Comment (smalluban) [reply src/background/onboarding.js:18]: As we decided, the onboarding is shown only once after a clean installation (or updated from the previous version - which has not yet set that value). I am closing the conversation.

File: src/pages/onboarding/index.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta name="theme-color" content="#00AEF0" />
+    <title>Welcome to Ghostery</title>
```
Comment (smalluban) [inline src/pages/onboarding/index.html:8]: I put that title here, but I am not sure if we want something else, as we don't translate it.

**Comment by smalluban** [comment]
The once-a-day open flow is implemented. I pushed and pulled translations. @philipp-classen If you have time, you can review the code.

File: src/background/communication.js [hunk 4]
```diff
@@ -56,24 +58,26 @@ async function send(protocol) {
   await updateSentAt();
 }
 
-(async function () {
+observe('terms', async (terms) => {
```
Comment (philipp-classen) [inline src/background/communication.js:61]: "terms" has the semantic of "terms accepted", right? If so, perhaps we can renamed it?

File: src/store/options.js [hunk 5]
```diff
@@ -34,10 +35,17 @@ const Options = {
       await chrome.storage.local.set({ options });
 
       // Send update message to another contexts (background page / panel / options)
-      await chrome.runtime.sendMessage({
-        action: UPDATE_OPTIONS_ACTION_NAME,
-        options,
-      });
+      try {
+        chrome.runtime.sendMessage({
```
Comment (philipp-classen) [inline src/store/options.js:39]: Is it intentional to drop the await here and make it fire-and-forget?

File: src/background/communication.js (hunk 4 above)
Comment (smalluban) [reply src/background/communication.js:61]: yes, it means terms accepted, but I think it is easy to reason about, as `terms: true` means, you have them, which means they are accepted. Why would you add an additional suffix?

File: src/store/options.js (hunk 5 above)
Comment (smalluban) [reply src/store/options.js:39]: Yes, we have to do this, as I discovered, that if the background process runs without another cotexts, this chrome.runtime.sendMessage()` throws an unexpected error (on one of the browsers, Safari or Chrome, I don't remember which). It is weird behavior, but the code should not break in that place, as then options are not set correctly. The call is only to notify other open contexts (in our case opened panel or options page, or the background process) that options have changed.

File: src/background/communication.js (hunk 4 above)
Comment (philipp-classen) [reply src/background/communication.js:61]: OK. Reading through it, it was not obvious to me that it is a flag and what it means.

**Pull request merged**